doc/kernel/usermode/usermode.rst

```diff
@@ -9,6 +9,142 @@ are defined, and how memory may be managed to support user mode threads.
 :ref:`lifecycle_v2`.
 
+Threat Model
+============
+
+For threads that have either started initially, or dropped down to user mode,
+we aim to protect against the following:
+
+- Access to memories outside their own stack or program text not specifically
+  granted, or incorrect access to memories that have an incompatible policy,
+  such as attempting to write to a read-only area.
+
+- Access to device drivers or kernel objects not specifically granted,
+  with the permission granularity on a per object or per driver instance basis.
+
+- Making kernel or driver API calls with incorrect parameters that would
+  otherwise cause a crash or corruption of data structures private to the
+  kernel. This includes:
+
+  - Using the wrong kernel object type.
+
+  - Using parameters outside of proper bounds or with nonsensical values.
+
+  - Passing memory buffers that the calling thread does not have sufficient
+    access to read or write, depending on the semantics of the API.
+
+  - Use of kernel objects that are not in a proper initialization state.
+
+- Ensure the detection and safe handling of user mode stack overflows.
+
+- Invoking system calls which are not built into the system due to kernel
+  configuration disabling them.
+
+We are specifically not protecting against the following attacks:
+
+- The kernel itself, and any threads that are executing in supervisor mode,
+  are assumed to be trusted.
+
+- The kernel build is assumed to be trusted. There is considerable build-time
+  logic for creating the tables of valid kernel objects, defining system calls,
+  and configuring interrupts. The .elf binary files that are worked with
+  during this process are all assumed to be trusted code.
+
+- Mistakes made in memory domain configuration done in kernel mode that
+  exposes private kernel data structures to a user thread. RAM for kernel
+  objects should always be configured as supervisor-only.
+
+- It is possible to make top-level declarations of user mode threads and
+  assign them permissions to kernel objects. In general, all C and header
+  files that are part of the kernel build producing zephyr.elf are assumed to
+  be trusted.
+
+- We do not protect against denial of service attacks through thread CPU
+  starvation. Zephyr has no thread priority aging and a user thread of a
+  particular priority can starve all threads of lower priority, and also other
+  threads of the same priority if time-slicing is not enabled.
+
+- There are build-time defined limits on how many threads can be active
+  simultaneously, after which creation of new user threads will fail.
+
+- Stack overflows for threads running in supervisor mode may be caught,
+  but the integrity of the system cannot be guaranteed.
+
+Example use-cases:
+
+- Protect against unintentional programming errors which would otherwise
+  silently corrupt the system.
+
+- Sandbox interpreters so malicious third-party code cannot compromise the
```

Comment:
We don't define sandbox interpreters that could prevent malicious third-party code
    from compromising the kernel or other threads.